experiences/homepage/homepage.test.js

```diff
@@ -0,0 +1,84 @@
+const { bootstrapExperience } = require('../../e2e/utils/bootstrap');
+const { scrollTo } = require('../../e2e/utils/scroll');
+const { touchCard } = require('../../e2e/utils/touch');
+const { isAtSnapPoint } = require('../../e2e/utils/snapPoints');
+const { isCardContentLoaded } = require('../../e2e/utils/cardContent');
+const { expect } = require('chai');
+const { getUrlFixture } = require('../../e2e/utils/fixtureUrl');
+const experience = require('./homepage.json');
+
+describe('homepage experience', function() {
+	let config,
+		fixture;
+	const fixtureUrl = getUrlFixture({
+		siteUrl: 'https://playground.marfeel.com/templates/article-example.html',
+		requestHostname: 'playground.marfeel.com',
+		technology: 'web',
+		experienceUrl: '/experiences/homepage/homepage.json'
+	});
+
+	it('setup', async function() {
+		config = experience;
+		fixture = {
+			url: fixtureUrl,
+			articleTitle: 'Article example'
+		};
+
+		await bootstrapExperience(browser, config, fixture);
+	});
+
+	it('card should render on scroll', async function() {
+		await scrollTo(browser, 400);
+
+		const firstCard = await browser.$(config.cards.homepage.cardSelector);
+
+		const firstCardExists = await firstCard.waitForExist({ timeout: 5000 });
+
+		expect(firstCardExists).equal(true);
+	});
+
+	it('card should have right content', async function() {
+		const rightContentLoaded = await isCardContentLoaded(browser,
+			config.cards.homepage.cardSelector,
+			config.cards.homepage.content);
+
+		expect(rightContentLoaded).equal(true);
+	});
+
+	it('card should be displayed in viewport at initial snap point', async()=>{
+		await scrollTo(browser, 800);
+
+		const firstCard = await browser.$(config.cards.homepage.cardSelector);
+
+		const firstCardIsInViewport = await firstCard.isDisplayedInViewport();
+
+		expect(firstCardIsInViewport).equal(true);
+
+		const isAtInitialSnapPoint = await isAtSnapPoint(browser,
+			config.cards.homepage.cardSelector,
+			config.cards.homepage.snapPoints.initial);
+
+		expect(isAtInitialSnapPoint).equal(true);
+	});
+
+	// fix minimise
+	// it("minimise card and should be at minimised snap point", async ()=>{
+	// 	await dragCardBy(browser, config.cards.homepage.cardSelector, 200)
+
+	// 	const isAtMinimisedSnapPoint = await isAtSnapPoint(browser,
+	// 		config.cards.homepage.cardSelector,
+	// 		config.cards.homepage.snapPoints.minimised)
+
+	// 	expect(isAtMinimisedSnapPoint).equal(true);
+	// });
+
+	it('activate card by click', async()=>{
+		await touchCard(browser, config.cards.homepage.cardSelector);
+
+		const isAtActiveSnapPoint = await isAtSnapPoint(browser,
+			config.cards.homepage.cardSelector,
+			config.cards.homepage.snapPoints.active);
+
+		expect(isAtActiveSnapPoint).equal(true);
+	});
+});
```

Comment:
Based on the [`flowcards.json`](https://github.com/Marfeel/flowcards-playground/blob/9f78fd6dd36d683d087c323e57377eb69b89407f/experiences/homepage/homepage.json#L1) for the homepage experience, I think we have several tests missing:

- `mode: modal`  that is, the blackout foreground appears when the Flowcard is `active` obscuring the content behind

- `button: {action: "setSnapPoint:initial"}` that is, when the Flowcard is open, and you scroll a little bit inside, a close button appears in the upper right corner, and if you click it, the Flowcard goes back to the `initial` snapPoint

- `infiniteScroll: true` that is, when I reach the end of the article, and I try to keep scrolling, the Flowcard starts going up, showing the homepage, in a ""infinite scroll like" experience



For me is ok to land this PR without this test, so we start testing everything together, but please, remember to complete the testing for the whole experience

Reply:
Yes, of course we are going to test the whole experience. This is just the initial POC. 
My idea was to land this on master and then work on additional test features, so we don't overload this PR more.